### ISSUE TITLE
top-level option for configuring btrfs multi-device volume

### ISSUE THREAD

**Issue by montchr** [nix-community/disko#261]
### Context

With the following devices on a Hetzner AX52:

```
/dev/nvme0n1
/dev/nvme1n1
/dev/sda
/dev/sdb
```

I am looking to implement a layout similar to the following example:

<https://github.com/tfc/nixos-anywhere-example/blob/fd710be5255931a64a9efb9c15c9b36e6011e7b4/two-raids-on-two-disks.nix>

In contrast to that layout, I would like to use BTRFS' native support for multi-device volumes instead of using mdadm.

## Problem

I would expect to be able to configure subvolumes for a multi-device BTRFS volume at `disko.devices.btrfs` (or some more-fitting name) in the same way that other volume-like domains e.g. `mdadm`, `zpool` do:

https://github.com/nix-community/disko/blob/5d9f362aecd7a4c2e8a3bf2afddb49051988cab9/lib/default.nix#L306-L334

## Question

Is it currently possible to configure multi-device BTRFS subvolumes using Disko?

**Comment by Mic92** [comment]
No this is not implemented just now, but feel free to implement it.
@Lassulus might have some guidelines on how to do this.

**Comment by phaer** [comment]
Pretty much the same feature is also supported for zfs user space tools, but Disko doesn't implement it for zfs neither so far.

**Comment by Mic92** [comment]
Well, you can great raids with zfs: https://github.com/nix-community/disko/blob/5d9f362aecd7a4c2e8a3bf2afddb49051988cab9/example/zfs.nix#L58

**Comment by phaer** [comment]
> Well, you can great raids with zfs:

Yes, that's what I am talking about here. But with standard zfs tools you can i.e. create a mirror (or raidz) on N disks and add a cache on a faster SSD, all in the same command. I.e. 

```
zpool create my-pool raidz /dev/by-name/disk-1 /dev/by-name/disk-2 /dev/by-name/disk-3 [..] /dev/by-name/disk-N cache /dev/by-name/ssd-1
```
I believe this isn't supported by disko at the moment and seems similar to the btrfs feature in terms of nixos module code? Pretty sure there's a thread where @Lassulus and me've talked about this already but i can't find that atm.

**Comment by stephenandary** [comment]
> No this is not implemented just now, but feel free to implement it. @Lassulus might have some guidelines on how to do this.

@Lassulus, after the holidays may I please have guidelines for this?

**Comment by iFreilicht** [comment]
> with standard zfs tools you can i.e. create a mirror (or raidz) on N disks and add a cache on a faster SSD, all in the same command. I believe this isn't supported by disko at the moment and seems similar to the btrfs feature in terms of nixos module code?

It is since #723. I would say an implementation for btrfs should follow a similar structure.
